mypy/checker.py

```diff
@@ -2945,7 +2946,11 @@ def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
                     "A type on this line", AnyType(TypeOfAny.special_form), s
                 )
             else:
-                self.msg.unimported_type_becomes_any("Type of variable", s.type, s)
+                self.msg.unimported_type_becomes_any(
+                    "Type of variable",
+                    s.type.item if isinstance(s.type, RequiredType) else s.type,
```

Comment:
Maybe it would be better to use some existing function to covert `RequiredType` to `ProperType`? Like `get_proper_type`? Or maybe something similar. Right now it feels too specific.

Reply:
`get_proper_type` is already being called—that’s where it’s crashing. It doesn’t convert `RequiredType`. Should it? That seems like a much bigger change though.

Reply:
I am not sure about it :)
cc @JukkaL and @ilevkivskyi

Reply:
This PR is not a fix, but a horrible dirty hack. I am actually going to revert another recent crash "fix" by @andersk and replace it with a proper one when I will have time.

Reply:
Sorry to have offended you!